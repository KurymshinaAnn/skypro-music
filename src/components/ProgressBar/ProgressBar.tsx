// import { ChangeEventHandler } from "react";
import styles from "./ProgressBar.module.css";

type ProgressBarType = {
  max: number | undefined;
  step: number;
  value: number | undefined;
  onChange: (value: any) => void;
  //   onChange: ChangeEventHandler;
};

export default function ProgressBar({
  max,
  value,
  step,
  onChange,
}: ProgressBarType) {
  const handleChange = (e: any) => {
    const newValue = e.target.value;
    onChange(newValue);
  };

  return (
    <input
      className={styles.styledProgressInput}
      type="range"
      min="0"
      max={max}
      value={value}
      step={step}
      onChange={handleChange}
    />
  );
}
