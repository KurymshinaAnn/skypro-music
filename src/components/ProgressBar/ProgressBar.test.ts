import renderer from "react-test-renderer";
import ProgressBar from "./ProgressBar";

it("Renders progress bar", () => {
  const mockCallback = jest.fn();
  const progressBar = renderer.create(
    ProgressBar({
      value: 0,
      max: 100,
      step: 5,
      onChange: mockCallback,
    })
  );
  let tree = progressBar.toJSON();
  expect(tree).toMatchSnapshot();
});
