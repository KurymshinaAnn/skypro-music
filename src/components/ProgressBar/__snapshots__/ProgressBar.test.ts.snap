// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders progress bar 1`] = `
<input
  className="styledProgressInput"
  max={100}
  min="0"
  onChange={[Function]}
  step={5}
  type="range"
  value={0}
/>
`;
